### PR TITLE
kucoin: fix spot order unmarshal bug

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Fixes bug when unmarshalling data, removes clientID -> clientOrderID mutation. 
* Adds time in force handling for spot orders

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — APPROVED** [review]
Ack due to no funds, but I have verified each change against the API docs

File: exchanges/kucoin/kucoin.go [hunk 1]
```diff
@@ -840,11 +841,8 @@ func (ku *Kucoin) PostOrder(ctx context.Context, arg *SpotOrderParam) (string, e
 	default:
 		return "", fmt.Errorf("%w %s", order.ErrTypeIsInvalid, arg.OrderType)
 	}
-	resp := struct {
-		OrderID string `json:"orderId"`
-		Error
-	}{}
-	return resp.OrderID, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, placeOrderEPL, http.MethodPost, kucoinPostOrder, &arg, &resp)
+	var resp SpotOrderResponse
```
Comment (thrasher-) [inline exchanges/kucoin/kucoin.go:844]: Curious as to why this was exported? We're still returning just the order ID rather than the struct resp

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes!

**Review by gloriousCode — APPROVED** [review]
tACKbert! Thanks for the additional improvements

**Pull request merged**